src/Domain/Tags.hpp

```diff
@@ -259,13 +256,40 @@ struct evaluate_compute_item<DirectionsTag, BaseComputeItem,
   }
 };
 
+// db::PrefixTag is only needed in the no-inheritance case, because if
+// we are inheriting off another interface tag we will get the
+// PrefixTag base class from that.
+template <typename DirectionsTag, typename Tag, typename TagType,
+          typename = std::nullptr_t>
+struct InterfaceBaseTag : db::PrefixTag, TagType {};
```

Comment:
I'm not clear on how these can be a base tag. If it's a base tag it must inherit off of `db::BaseTag` and not `db::SimpleTag` or `db::ComputeTag`. To me it looks like it's currently confused between being a base tag and a more concrete tag. Could you add a comment explaining how these aren't violating the base tag requirements?

Reply:
I meant it in the sense of a base class (and also the "base tag mechanism" that it is implementing).  I am not aware of a term for the part of the base tag mechanism that isn't related to `db::BaseTag`.  It is actually not possible to use a `db::BaseTag`-style base tag with interface items because interface items require `db::item_type` to work.

Reply:
Ah, I understand, sorry. Yes, we should try and find a way to differentiate those :( I'm currently not coming up with any name that isn't horrible and super vague...

Could the `InterfaceBaseTag` then always be a simple tag? The design goal was that you could have compute tags inherit off of simple tags, in which case the return type lookup is more efficient at compile time. Or is there something that currently relies of the base class tag also being marked as a compute tag? I now that some parts of the DataBox code, specifically item_type, avoids looking up derived tags if the tag is not a `db::BaseTag` and I'm worried that this might cause unexpected and confusing errors in the future.

Reply:
I can't think of any realistic cases where this would try to use a compute tag as a base class.  You could confuse it if you tried, but I'm not coming up with anything that would arise naturally.

Most of this mess if from avoiding compiler warnings from inheriting from the tag markers twice.  I think I'll just give up and mark them as virtual.